logger/sdkLogger.go

```diff
@@ -0,0 +1,336 @@
+	level Level,
+	request *http.Request,
+	logRequest RequestLoggerConfiguration) {
+
+	if logRequest.body {
+		a.logger.Log(level, fmt.Sprintf("Request body %v", request.Body),
+			map[string]any{"body": request.Body},
+		)
+	}
+}
+
+func (a *SdkLogger) _applyLogResponseOptions(level Level, response *http.Response) {
+	a._applyLogResponseHeaders(
+		level,
+		response,
+		a.loggingOptions.response,
+	)
+
+	a._applyLogResponseBody(
+		level,
+		response,
+		a.loggingOptions.response,
+	)
+}
+
+func (a *SdkLogger) _applyLogResponseHeaders(
+	level Level,
+	response *http.Response,
+	logResponse MessageLoggerConfiguration) {
+
+	logHeaders := logResponse.headers
+	headersToInclude := logResponse.includeHeaders
+	headersToExclude := logResponse.excludeHeaders
+	headersToWhitelist := logResponse.whitelistHeaders
+
+	if logHeaders {
+		var headersToLog = a._extractHeadersToLog(
+			headersToInclude,
+			headersToExclude,
+			headersToWhitelist,
+			response.Header,
+		)
+
+		a.logger.Log(level, fmt.Sprintf("Response headers %v", headersToLog),
+			map[string]any{"headers": headersToLog},
+		)
+	}
+}
+
+func (a *SdkLogger) _applyLogResponseBody(
+	level Level,
+	response *http.Response,
+	logResponse MessageLoggerConfiguration) {
+
+	if logResponse.body {
+		a.logger.Log(level, fmt.Sprintf("Response body %v", response.Body),
+			map[string]any{"body": response.Body},
+		)
+	}
+}
+
+const CONTENT_TYPE_HEADER = "content-type"
+const CONTENT_LENGTH_HEADER = "content-length"
+
+func (a *SdkLogger) _getContentType(headers http.Header) string {
+	var contentType string = ""
+	if len(headers) > 0 {
+		contentType = headers.Get(CONTENT_TYPE_HEADER)
+	}
+	return contentType
+}
+
+func (a *SdkLogger) _getContentLength(headers http.Header) string {
+	var contentLength string = ""
+	if len(headers) > 0 {
+		contentLength = headers.Get(CONTENT_LENGTH_HEADER)
+	}
+	return contentLength
+}
+
+func (a *SdkLogger) _removeQueryParams(url string) string {
+	if strIndex := strings.Index(url, "?"); strIndex != -1 {
+		return url[:strIndex]
+	}
+	return url
+}
+
+func (a *SdkLogger) _extractHeadersToLog(
+	headersToInclude []string,
+	headersToExclude []string,
+	headersToWhitelist []string,
+	headers http.Header) http.Header {
+	filteredHeaders := http.Header{}
+	if !(len(headers) > 0) {
+		return http.Header{}
+	}
+
+	if len(headersToInclude) > 0 {
+		filteredHeaders = a._includeHeadersToLog(
+			headers,
+			filteredHeaders,
+			headersToInclude,
+		)
+	} else if len(headersToExclude) > 0 {
+		filteredHeaders = a._excludeHeadersToLog(
+			headers,
+			filteredHeaders,
+			headersToExclude,
+		)
+	} else {
+		filteredHeaders = headers
+	}
+
+	return a._maskSenstiveHeaders(filteredHeaders, headersToWhitelist)
+}
+
+func (a *SdkLogger) _includeHeadersToLog(
+	headers, filteredHeaders http.Header,
+	headersToInclude []string) http.Header {
+	// Filter headers based on the keys specified in includeHeaders
+	for _, name := range headersToInclude {
+		val, ok := headers[name]
```

Comment:
found another instance of case sensitive header extraction. Headers extraction should always be non sensitive.

Reply:
updated the code to match non-sensitive headers matching